driver/call.py

```diff
@@ -36,6 +36,8 @@ def fail(exception, exitcode):
                 fail(err, returncodes.DRIVER_CRITICAL_ERROR)
             except ValueError as err:
                 fail(err, returncodes.DRIVER_INPUT_ERROR)
+            except TypeError as err:
+                fail(err, returncodes.DRIVER_CRITICAL_ERROR)
```

Comment:
There doesn't seem to be a logical order to these exceptions. For example, they could be grouped by severity (in which case I'd also combine the ones that go together, using the tuple `(OSError, TypeError)`). But also TypeError is likely a bug and OSError is likely something wrong with the environment in which the driver is run, so it's a pity we can't distinguished them. But it's probably not worth adding a new exit code over this.

More generally, I'm not so sure if we should add exception handling that catches specific *programming* errors; is it really that much more likely that the next time we introduce a bug it will be a `TypeError` and not something else? I didn't mind the previous behaviour with the generic message where you have to dig a bit deeper if and when an error occurs. I think the existing message (if I understood this correctly) about an exception in a preexec function is enough for this.

If you disagree, I'd rather have a catch-all clause here (`except Exception as err`) than catching only `TypeError`.

Reply:
I have no strong opinion about this. I removed the lines again.